sdks/python/apache_beam/ml/transforms/base_test.py

```diff
@@ -443,6 +442,26 @@ def test_handler_on_multiple_columns(self):
           equal_to(expected_data),
       )
 
+  def test_handler_on_columns_not_exist_in_input_data(self):
+    data = [
+        {
+            'x': "Hello world", 'y': "Apache Beam"
```

Comment:
> For each subsequent record, before adding it to the dictionary check that it contains all the same fields and doesn't contain extra fields. Throw if it doesn't satisfy these conditions, else add it to the dictionary.

Alternately, we could have some sort of dummy value that we use as a placeholder for empty fields and only throw if we're performing the operation. That would probably be more work than we want to do right before the release cut, but the idea would basically be given data:

```
data = [
{
   'x': 'foo', 'y': 'bar', 'z': 'baz'
},
{
   'x': 'foo2', 'y': 'bar2'
}
]
```

`MLTransform(...).with_transform(['x', 'y'])` doesn't need to throw, it would return:

```
{
   'x': fn('foo'), 'y': fn('bar'), 'z': 'baz'
},
{
   'x': fn('foo2'), 'y': fn('bar2')
}
]
```

but `MLTransform(...).with_transform(['x', 'y', 'z'])` would throw because it would want to apply `fn(record.z)` on the second record.

My vote would be to just tighten the error checking as mentioned in my first comment, but its something we could consider.

Reply:
tensorflow-transform throws the error for this kind of  example.  Maybe we should follow same pattern? may be not? let me think about it.
```
data = [
{
   'x': 'foo', 'y': 'bar', 'z': 'baz'
},
{
   'x': 'foo2', 'y': 'bar2'
}
]
```

Reply:
Yeah, at least for now I think we should throw and be consistent (and ideally document that behavior).

Reply:
(especially since release cut is today and that's a relatively straightforward change)

Reply:
I made changes according to
1. Add the first record to the dictionary
2. For each subsequent record, before adding it to the dictionary check that it contains all the same fields and doesn't contain extra fields. Throw if it doesn't satisfy these conditions, else add it to the dictionary.